controllers/groupsync_controller_test.go

```diff
@@ -0,0 +1,127 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	userv1 "github.com/openshift/api/user/v1"
+	redhatcopv1alpha1 "github.com/redhat-cop/group-sync-operator/api/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	// +kubebuilder:scaffold:imports
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+
+var _ = Describe("GroupSyncController controller", func() {
+	const (
+		preexistingGroup = "preexisting-group"
+		namespace        = "default"
+	)
+
+	BeforeEach(func() {
+		ctx := context.Background()
+
+		By("By having pre-existing openshift groups")
+		ocpGroup := &userv1.Group{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: preexistingGroup,
+			},
+		}
+		Expect(k8sClient.Create(ctx, ocpGroup)).Should(Succeed())
+
+	})
+
+	AfterEach(func() {
+		ctx := context.Background()
+
+		k8sClient.DeleteAllOf(ctx, &redhatcopv1alpha1.GroupSync{}, client.InNamespace(namespace))
+		k8sClient.DeleteAllOf(ctx, &userv1.User{})
+		k8sClient.DeleteAllOf(ctx, &userv1.Group{})
+	})
+
+	It("It should sync groups from the provider", func() {
```

Comment:
For Ginkgo tests you don't need to add the `It` or `By` as in the output that should be added already.  But that's a very minor nitpick :)
```suggestion
	It("should sync groups from the provider", func() {
```